lazyflow/distributed/TaskOrchestrator.py

```diff
@@ -0,0 +1,78 @@
+from mpi4py import MPI
+from threading import Thread
+import enum
+from typing import Generator, TypeVar, Generic, Callable, Tuple
+import uuid
+
+END = "END-eb23ae13-709e-4ac3-931d-99ab059ef0c2"
```

Comment:
could you please add some comments here on the lingo used? What is a datum? What is a tag. So that ppl can make sense of this code without going through an mpi tutorial. (is 13 a special number?). Thx